### ISSUE TITLE
Unable to manually type a negative value in the input

### ISSUE THREAD

**Issue by tombertrand** [vlad-ignatov/react-numeric-input#50]
Hey, perhaps I'm doing something wrong, here is my configuration

```javascript
<NumericInput
    ref="offsetLeft"
    min={-100}
    max={100}
    step={1}
    onChange={this.setLabelLeftOffset}
    value={this.state.offsetLeft}
    size={4}
    style={{
        input: {
            marginRight: 3,
            paddingRight: 4,
            textAlign: 'right',
        },
        btnUp: {
            display: 'none',
        },
        btnDown: {
            display: 'none',
        },
    }}
/>
```

I'm trying to type a negative value inside the input starting by the character `-` (minus), but it does not register. I'm only allowed to go to the negative value if I use the arrow down or if a number is already typed.

**Comment by shahyar** [comment]
[I forked it and fixed it for you in `2.0.9`](https://github.com/shahyar/react-numeric-input). This isn't on NPM, so you'll have to target the specific git hash in your `package.json`:

`"react-numeric-input": "https://github.com/shahyar/react-numeric-input.git#b01f1f9"`

Only code change to be aware of is that because you pass `value={this.state.offsetLeft}`, which you presumably set in `onChange={this.setLabelLeftOffset}`, make sure you store the second argument of `onChange(valueAsNumber, valueAsString)`, since `valueAsNumber` will drop the prefixed `+`, `-`, and `.`.


TU ES BON!!

**Comment by tombertrand** [comment]
Amazing! 💃

**Comment by vlad-ignatov** [comment]
This should be fixed in 2.0.9.

**Comment by mpeng** [comment]
Please help take a look. This is an important requirement for my app. My users often want to enter a negative number manually using keyboard without using arrows first.

i.e. the initial value is "0.0", then I am unable to enter "-1.0". The only way I can enter "-" is after I change the number from "0.0" to "1.0", then then I am able to move cursor to the beginning and add "-" sign.

I did not see the fix in the version that I am using, which is:
"react-numeric-input": "^2.2.3"

Below is my code, as you can see, once the this.props.callbackFromParent(a); is added, it won't take "-" sign without changing the default value from 0.0 to something like "2.0". However if I comment out this.props.callbackFromParent(a); then I am able to enter "-" before "0.0", everything is fine. I could never figure out why the callback method makes this difference.

  test = (a) => {
    console.log(a);
    **// ToDo Why with below line, the minus "-" sign can not be entered.**
    this.props.callbackFromParent(a);
  }

  render() {
    const { o } = this.state;

    return (
    <NumericInput
        value={o}
        format={o => o}
        onChange={this.test}
        precision={this.props.precision}
        size={8}
        step={this.props.step}
        mobile={false}
        min={this.props.min}
        max={this.props.max}
